subprojects/core-api/src/main/java/org/gradle/api/file/ContentFilterable.java

```diff
@@ -78,7 +78,7 @@
      * @param transformer to implement line based filtering
      * @return this
      */
-    ContentFilterable filter(Transformer<String, String> transformer);
+    ContentFilterable filter(Transformer<String, /*@Nullable*/ String> transformer);
```

Comment:
I think we should at least leave a TODO marker there so we can annotate the type once we migrate to Java 8.